firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -296,27 +338,61 @@ AppDistributionReleaseInternal getCachedLatestRelease() {
         AlertDialog.BUTTON_POSITIVE,
         context.getString(R.string.update_yes_button),
         (dialogInterface, i) ->
-            // show download progress in notification manager
+        // show download progress in notification manager
+        {
+          synchronized (updateTaskLock) {
             updateApp(true)
-                .addOnSuccessListener(unused -> updateAlertDialogTask.setResult(null))
-                .addOnFailureListener(updateAlertDialogTask::setException));
+                .addOnProgressListener(
+                    progress -> cachedUpdateToLatestReleaseTask.updateProgress(progress))
+                .addOnSuccessListener(unused -> cachedUpdateToLatestReleaseTask.setResult())
+                .addOnFailureListener(cachedUpdateToLatestReleaseTask::setException);
+          }
+        });
 
     alertDialog.setButton(
         AlertDialog.BUTTON_NEGATIVE,
         context.getString(R.string.update_no_button),
         (dialogInterface, i) -> {
           dialogInterface.dismiss();
-          updateAlertDialogTask.setException(
-              new FirebaseAppDistributionException(
-                  Constants.ErrorMessages.UPDATE_CANCELED,
-                  FirebaseAppDistributionException.Status.INSTALLATION_CANCELED));
+          synchronized (updateTaskLock) {
+            cachedUpdateToLatestReleaseTask.updateProgress(
+                UpdateProgress.builder()
+                    .setApkFileTotalBytes(UNKNOWN_RELEASE_FILE_SIZE)
+                    .setApkBytesDownloaded(UNKNOWN_RELEASE_FILE_SIZE)
+                    .setUpdateStatus(UpdateStatus.UPDATE_CANCELED)
+                    .build());
+            cachedUpdateToLatestReleaseTask.setException(
+                new FirebaseAppDistributionException(
+                    Constants.ErrorMessages.UPDATE_CANCELED,
+                    FirebaseAppDistributionException.Status.INSTALLATION_CANCELED));
+          }
         });
 
     alertDialog.show();
-    return updateAlertDialogTask.getTask();
+    synchronized (updateTaskLock) {
+      return cachedUpdateToLatestReleaseTask;
+    }
   }
 
   void setInstallationResult(int resultCode) {
     this.updateAppClient.setInstallationResult(resultCode);
   }
+
```

Comment:
might be worth renaming this to be more specific for the updateToLatestRelease task

Reply:
Agreed. Naming changes are planned to go into a separate PR as of now.